clickhouse/exec.sh

```diff
@@ -25,36 +25,92 @@ ch_active || exit 1
 # tail -n+2 data/G1_1e7_1e2_0_0.csv | clickhouse-client --query="INSERT INTO G1_1e7_1e2_0_0 SELECT * FROM input('id1 Nullable(String), id2 Nullable(String), id3 Nullable(String), id4 Nullable(Int32), id5 Nullable(Int32), id6 Nullable(Int32), v1 Nullable(Int32), v2 Nullable(Int32), v3 Nullable(Float64)') FORMAT CSV"
 
 # tune CH settings and load data
+sudo touch '/var/lib/clickhouse/flags/force_drop_table' && sudo chmod 666 '/var/lib/clickhouse/flags/force_drop_table'
 clickhouse-client --query 'DROP TABLE IF EXISTS ans'
 echo '# clickhouse/exec.sh: creating tables and loading data'
+# set ClickHouse parallelism at half of virtual cores
+THREADS=$(($(nproc --all) /2))
+HAS_NULL=$(clickhouse-client --query "SELECT splitByChar('_','$SRC_DATANAME')[4]>0 FORMAT TSV")
+IS_SORTED=$(clickhouse-client --query "SELECT splitByChar('_','$SRC_DATANAME')[5]=1 FORMAT TSV")
+ON_DISK=0
+
 if [ $1 == 'groupby' ]; then
   clickhouse-client --query "DROP TABLE IF EXISTS $SRC_DATANAME"
-  clickhouse-client --query "CREATE TABLE $SRC_DATANAME (id1 Nullable(String), id2 Nullable(String), id3 Nullable(String), id4 Nullable(Int32), id5 Nullable(Int32), id6 Nullable(Int32), v1 Nullable(Int32), v2 Nullable(Int32), v3 Nullable(Float64)) ENGINE = MergeTree() ORDER BY tuple();"
+  if [ $HAS_NULL -eq 1 ]; then
+    if [ $IS_SORTED -eq 1 ]; then
+      clickhouse-client --query "CREATE TABLE $SRC_DATANAME (id1 LowCardinality(Nullable(String)), id2 LowCardinality(Nullable(String)), id3 Nullable(String), id4 Nullable(Int32), id5 Nullable(Int32), id6 Nullable(Int32), v1 Nullable(Int32), v2 Nullable(Int32), v3 Nullable(Float64)) ENGINE = MergeTree() ORDER BY (id1,id2,id3,id4,id5,id6,v1,v2,v3);"
```

Comment:
sortedness is AFAIR only by ID cols, not v*

Reply:
I can change if you think it makes  more sense. IMO it should not help at all as first columns already define order

Reply:
yes, but it is somehow misleading now by treating all columns the same. ID columns are dimensions, V columns are measures.